### PR TITLE
Add toMultimap() method to Headers class

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/Headers.java [hunk 1]
```diff
@@ -130,6 +131,20 @@ public Builder newBuilder() {
     return result.toString();
   }
 
+  public Map<String, List<String>> toMultimap() {
+    HashMap<String, List<String>> result = new HashMap<String, List<String>>();
+    for (int i = 0, size = size(); i < size; i++) {
+      if (result.containsKey(name(i))) {
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/Headers.java]: You can avoid the multiple map lookups here by doing:
```java
String name = name(i);
List<String> values = result.get(name);
if (values == null) {
  values = new ArrayList<>(2);
  result.put(name, values);
}
values.add(value(i));
```

File: okhttp/src/main/java/com/squareup/okhttp/Headers.java [hunk 2]
```diff
@@ -130,6 +131,20 @@ public Builder newBuilder() {
     return result.toString();
   }
 
+  public Map<String, List<String>> toMultimap() {
+    HashMap<String, List<String>> result = new HashMap<String, List<String>>();
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/Headers.java]: As a general rule we always use `LinkedHashMap`, and in this case you'll get predictable iteration order which corresponds to the first occurrence of each header.

File: okhttp/src/main/java/com/squareup/okhttp/Headers.java [hunk 3]
```diff
@@ -130,6 +131,20 @@ public Builder newBuilder() {
     return result.toString();
   }
 
+  public Map<String, List<String>> toMultimap() {
+    LinkedHashMap<String, List<String>> result = new LinkedHashMap<String, List<String>>();
+    for (int i = 0, size = size(); i < size; i++) {
+      String name = name(i);
+      List<String> values = result.get(name);
+        if (values == null) {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Headers.java]: nit: I don't think this needs to be indented

File: okhttp/src/main/java/com/squareup/okhttp/Headers.java [hunk 4]
```diff
@@ -130,6 +131,20 @@ public Builder newBuilder() {
     return result.toString();
   }
 
+  public Map<String, List<String>> toMultimap() {
+    LinkedHashMap<String, List<String>> result = new LinkedHashMap<String, List<String>>();
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Headers.java]: nit: by convention we use interface type on LHS, like this:

```
Map<String, List<String>> result = new LinkedHashMap<>();
```

**Comment by swankjesse** [comment]
Sorry for the parade of nitpicks. One last one – do you mind doing a simple test case? HeadersTest is already ready to go.

**Comment by swankjesse** [comment]
Oh and one one more nitpick: please use http://rebaseandsqua.sh so we get only one commit in our git history!

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/HeadersTest.java [hunk 5]
```diff
@@ -302,4 +302,14 @@
     } catch (IllegalArgumentException expected) {
     }
   }
+
+  @Test public void toMultimapGroupsHeaders() {
+    Headers headers = Headers.of(
+        "cache-control", "no-cache",
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/HeadersTest.java]: dig.

File: okhttp/src/main/java/com/squareup/okhttp/Headers.java [hunk 6]
```diff
@@ -130,6 +131,20 @@ public Builder newBuilder() {
     return result.toString();
   }
 
+  public Map<String, List<String>> toMultimap() {
+    Map<String, List<String>> result = new LinkedHashMap<String, List<String>>();
+    for (int i = 0, size = size(); i < size; i++) {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Headers.java]: Nice.

**Comment by swankjesse** [comment]
Thank you!

**Pull request merged**